src/sentry/tasks/statistical_detectors.py

```diff
@@ -268,9 +284,7 @@ def _detect_transaction_trends(
     start = start - timedelta(hours=1)
     start = start.replace(minute=0, second=0, microsecond=0)
     end = start + timedelta(hours=1)
-    all_transaction_payloads = query_transactions(
-        org_ids, project_ids, start, end, TRANSACTIONS_PER_PROJECT
-    )
+    all_transaction_payloads = get_all_transaction_payloads(org_ids, project_ids, start, end)
```

Comment:
the only thing is that we're passing all org ids so they'll be in the query. maybe I can only pass org ids associated with the projects ids

Reply:
I don't think this is a big issue. Since we're passing a subset of the project ids as well, the data skipping index in clickhouse should handle this nicely.